### ISSUE TITLE
Add embed snippet to Thimble publishing dialog

### ISSUE THREAD

**Issue by ldecoursy** [mozilla/thimble.mozilla.org#1543]
Can we add a "prefab" embed snippet (like YouTube and others) so when publishing a Thimble remix you can grab an embed code?

**Comment by gideonthomas** [comment]
I think this needs design first!

**Comment by humphd** [comment]
There are two parts to this:

1. UI for the user to get the snippet.  @flukeout can you provide some UI/design for this bit?
2. The actual HTML for an `iframe` wrapped version of the user's page on S3, for example `<iframe src="https://thimbleprojects.org/humphd/191373/"></iframe>`

**Comment by flukeout** [comment]
If someone takes this on I'll gladly do the design. I think we should only show it on the desktop version of the remix button details bar.

**Comment by Pomax** [comment]
@cgsingh I'm seeing commits in this issue from your repository, does that mean you are currently working on trying to solve this issue? If so, I would encourage you to communicate that here so we know you're on the job, and can help with any questions in the right place =)

**Comment by cgsingh** [comment]
@Pomax Oh dear. I accidentally referenced 1543 instead of 1534 on my last commit to my only pull request. Sorry, bad typo.

**Comment by humphd** [comment]
@cgsingh you can fix it like this, for future:

```
git commit --amend -m "Your new commit message here..."
git push origin branch-name -f
```

This will update the commit message on your last commit.

**Comment by DorelBarbu** [comment]
Hello! I have some ideas. If I understood correctly,  we want a button that when it's pressed it generates a little pop-up with a text field that contains the html code for an iframe. Obtaining the code it's a matter of concatenating strings, because it's not that complicated. Besides the text field, it would be nice to add some customizable options than can be chosen from drop-down lists or simply checkboxes (for example to chose the dimensions of the iframe), and update the html code accordingly. We could add a little button that copies the text to the clipboard so that the user doesn't have to manually select all the text in the field and right click copy. What do you guys think? Am I on the right track?

**Comment by flukeout** [comment]
Assigning to @DorelBarbu - yeah, the things you mentioned all sound good. If you want to tackle the design as well, then let's start with that. We'll use the design phase to figure out the features you want. 

Pay attention also to the fact that the Remix button has different styling depending on mobile and desktop modes. We might not want to even have this functionality on mobile, as it's doubtful that you'll be creating a webpage or editing a document while on your mobile device.

**Comment by DorelBarbu** [comment]
Hello! If I understood correctly we need to place the button in this green remix bar.
![remix button](https://cloud.githubusercontent.com/assets/24645887/24057498/3530cc48-0b50-11e7-8200-6d73917d1099.png)

But as @gideonthomas  it's a little crowded there. @flukeout how should we do it?

**Comment by flukeout** [comment]
Yeah, that's where it will go.. we can try a couple of approaches:

* Make this box taller and include the iframe code below the description
* Add a "Embed" button beside the Remix button
  * If you click it it reveals the UI with the iframe code
  * It can be hidden when the remix bar is not expanded

Let me know if you want more specific guidance or if you'd like to take a shot at a design first?

**Comment by DorelBarbu** [comment]
I would go with the second approach because it seems more interesting and resembles more with what I had in mind. I would like to take a shot at the design, but I don't know where to start. I believe the "Embed" button should keep the same style as the "Remix" button. Question is, where should it be placed? Left, right, above, below of the "Remix" button? I would say below.

**Comment by flukeout** [comment]
I think maybe to the left of the remix button? Give that a try maybe. The label for it could just be something like ``</>`` I think that's what some other sites do.

**Comment by flukeout** [comment]
But, as a start, you can just make it work with a simple text link so we can build out the functionality and adjust styling as we go.

**Comment by flukeout** [comment]
In re-reading the original request, I just realized that it was meant as an option within the publish dialog itself, and not on the remix bar.

I think both are useful, but maybe I'll re-file both issues.

**Comment by flukeout** [comment]
Hey @DorelBarbu are you able to adapt your work to add the embed snippet to the Publish dialog instead?

![image](https://cloud.githubusercontent.com/assets/25212/24306917/7cf3f1ac-107f-11e7-93af-f79a54146015.png)

**Comment by DorelBarbu** [comment]
Sure thing, @flukeout. I've been a little busy this week but I will fix it soon. Thanks!

**Comment by DorelBarbu** [comment]
So, let's get this done. @flukeout , where should the button be positioned?
/home/dorel/Pictures/publish.png
What do you say? Would it be good to put it to the right of the publish button?

**Comment by flukeout** [comment]
Hey @DorelBarbu - welcome back! If you see my comment above you'll see I made some designs for this UI. We can't show the embed code until the project is published and has a live URL.

**Comment by DorelBarbu** [comment]
Good to see you too, @flukeout. I’m back indeed.  So, what exactly should I do now that  the design is done (looks great btw)? And another question: Did you make the design code public?

**Comment by flukeout** [comment]
I haven't created any markup for the design above yet. If you want to get started with a PR that shows the embed code, then I can add some markup around it with the tabs after. Let me know if that works for you.

**Comment by DorelBarbu** [comment]
Sure, @flukeout . How should I show the embed code for this stage? In an alert box, or how?

**Comment by flukeout** [comment]
Hi @DorelBarbu the UI I mocked up above should go into the Publish dialog, something like this...

![image](https://cloud.githubusercontent.com/assets/25212/24810294/c55380dc-1b76-11e7-8c13-86a21ed5a9fa.png)

So, when a user project is published, there is a little UI with two tabs that has two options:

* Display the link to the published project
* Display the embed code that they can use to add it to an HTML page

Is that more clear? Cheers!

**Comment by DorelBarbu** [comment]
Hello @flukeout , yes I understood how should the UI look. But what I still don't get is what should I do at this stage. I understood that you'll add the mark up later and that I should do a prototype. What should my prototype do?

**Comment by flukeout** [comment]
Hey @DorelBarbu - sorry for the confusion. I was hoping that you would start working inside of the ``view/editor/publish.html`` file and start to add the code that the user can see to embed an ``iframe`` tag. Once you have something basic working, you can create a PR and we can continue the work together there. Let me know if that helps.

**Comment by humphd** [comment]
This seems to have stalled.  I'm going to put it back into the general pool of bugs.  Also related to this is #2258.

**Comment by Fatehsandhu** [comment]
@humphd glad to see you here!  I'm a student at Seneca college learning open source, and I was hoping to work on this bug for my course.  If no one else is currently working on it, I'd like to give it a try.

**Comment by humphd** [comment]
Sounds good @Fatehsandhu, it's yours.  Please see the discussion above, and ask questions as you go.  @flukeout can help with the look-and-feel for what we'll need to do here.

**Comment by Fatehsandhu** [comment]
@humphd I will get on it right away, thanks a lot

**Comment by Fatehsandhu** [comment]
@humphd @flukeout I have a rough design going here. I was wondering if I could get some help in generating the iFrame code?

<img width="571" alt="screen shot 2017-10-13 at 18 48 16" src="https://user-images.githubusercontent.com/10791771/31569229-4fb237cc-b047-11e7-9316-7f19f101def9.png">

**Comment by Obtin** [comment]
@humphd @flukeout Is the issue resolved or anybody else is working on it ? If its not the case, I would like to work on this issue.
Thank you.

**Comment by Fatehsandhu** [comment]
@obtin I was working on it but I keep getting some sorts of errors when I create a pull request. 
All my code is here --> https://github.com/Fatehsandhu/thimble.mozilla.org
Happy New Years!
